File: Source/Easy_Reverse_Linked_List.cs
public class ListNode
{
    public int val;
    public ListNode next;
    
    public ListNode(int val=0, ListNode next=null)
    {
        this.val = val;
        this.next = next;
    }
}

public class Solution
{
    public ListNode Reverse_Linked_List(ListNode head)
    {
        ListNode next = null;
        ListNode current = head;
        ListNode previous = null;

        while(current != null)
        {
            next = current.next;
            current.next = previous;
            previous = current;
            current = next;
        }

        return previous;
    }
}

File: Source/Easy_Reverse_Linked_List.cpp
struct ListNode
{
	int val;
	ListNode *next;
	ListNode() : val(0), next(nullptr) {}
	ListNode(int x) : val(x), next(nullptr) {}
	ListNode(int x, ListNode *next) : val(x), next(next) {}
};


ListNode* Reverse_Linked_List(ListNode* head)
{
	ListNode* current = head;
	ListNode* next = nullptr;
	ListNode* previous = nullptr;

	while(current != nullptr)
	{
		next = current->next;
		current->next = previous;
		previous = current;
		current = next;
	}

	return previous;
}